Libraries/Image/RCTImageLoader.m

```diff
@@ -417,8 +420,16 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
 
     // Download image
     RCTNetworkTask *task = [_bridge.networking networkTaskWithRequest:request completionBlock:^(NSURLResponse *response, NSData *data, NSError *error) {
-      if (error) {
-        completionHandler(error, nil);
+      if (error || !response || !data) {
+        NSError *someError = nil;
+        if (error) {
+          someError = error;
+        } else if (!response) {
+          someError = RCTErrorWithMessage(@"Response metadata error");
+        } else {
+          someError = RCTErrorWithMessage(@"Unknown image download error");
```

Comment:
Here is a crash stack of this: https://github.com/facebook/react-native/issues/6952
Before the processResponse code, it would crash at `[[NSCachedURLResponse alloc] initWithResponse:response ...]` when response is nil.